### PR TITLE
farcaster tagging fix

### PR DESCRIPTION
**Pull request by sktbrd**
DRAFT: Dont push -> testing in production

### CODE REVIEW

File: src/common/components/molecules/Modal.tsx [hunk 1]
```diff
@@ -27,7 +27,7 @@ const Modal = ({
   showClose = true,
   overlay = true,
 }: ModalProps) => (
-  <Dialog.Root open={open} onOpenChange={setOpen} modal={focusMode}>
```
Comment (sktbrd) [inline src/common/components/molecules/Modal.tsx:30]: This focus mode is causing the Modal to close itself when the focus is shifted to the mentionList component

File: src/fidgets/farcaster/components/CreateCast.tsx [hunk 2]
```diff
@@ -299,11 +425,11 @@ const CreateCast: React.FC<CreateCastProps> = ({
             <EditorContent
               editor={editor}
               autoFocus
-              className="w-full h-full min-h-[150px] opacity-80 opacity-80"
+              className="w-full h-full min-h-[150px] opacity-80"
             />
             <div className="z-50">
               <EmbedsEditor
-                embeds={[]}
+                embeds={embeds}
```
Comment (sktbrd) [inline src/fidgets/farcaster/components/CreateCast.tsx:409]: We need to review those embeds editors and previews

File: src/fidgets/farcaster/components/mentionList.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
+"use client";
```
Comment (sktbrd) [inline src/fidgets/farcaster/components/mentionList.tsx:1]: Had to bring this code over so we have more control of the component

**Comment by willyogo** [comment]
I think this is ready to ship. Possible there are still a few edge cases, but it's definitely better than current state in prod, and the only way to find the edge cases is to test in prod.
